actor/builtin/builtin.go

```diff
@@ -12,16 +14,58 @@ import (
 	cid "github.com/ipfs/go-cid"
 )
 
-// Actors is list of all actors that ship with Filecoin.
-// They are indexed by their CID.
-var Actors = map[cid.Cid]exec.ExecutableActor{}
+type codeVersion struct {
+	code    cid.Cid
+	version uint64
+}
+
+type Actors struct {
+	actors map[codeVersion]exec.ExecutableActor
+}
+
+func (ba Actors) GetBuiltinActorCode(code cid.Cid, version uint64) (exec.ExecutableActor, error) {
+	if !code.Defined() {
+		return nil, fmt.Errorf("missing code")
+	}
+	actor, ok := ba.actors[codeVersion{code: code, version: version}]
+	if !ok {
+		return nil, fmt.Errorf("unknown code: %s", code.String())
+	}
+	return actor, nil
+}
+
+type BuiltinActorsBuilder struct {
+	actors map[codeVersion]exec.ExecutableActor
+}
 
-func init() {
-	// Instance Actors
-	Actors[types.AccountActorCodeCid] = &account.Actor{}
-	Actors[types.StorageMarketActorCodeCid] = &storagemarket.Actor{}
-	Actors[types.PaymentBrokerActorCodeCid] = &paymentbroker.Actor{}
-	Actors[types.MinerActorCodeCid] = &miner.Actor{}
-	Actors[types.BootstrapMinerActorCodeCid] = &miner.Actor{Bootstrap: true}
-	Actors[types.InitActorCodeCid] = &initactor.Actor{}
+func NewBuiltinActorsBuilder() *BuiltinActorsBuilder {
+	return &BuiltinActorsBuilder{actors: map[codeVersion]exec.ExecutableActor{}}
 }
+
+func BuiltinActorsExtender(actors Actors) *BuiltinActorsBuilder {
```

Comment:
This is a bit clunky. How about an `AddAll` method instead so the call is `builtin.NewBuilder().AddAll(actors).Build()`

Or just rename this method to `Extend()`.